src/semantics/expression_builder.ts

```diff
@@ -1270,6 +1272,9 @@ export function newCastValue(
                     );
                 }
             }
+            // iff r-value is empty.
+            // when the 'meta' of r-value is modified, the 'typeId' of r-value also needs to be modified.
+            if (isRValEmpty) from_value.type.typeId = type.typeId;
```

Comment:
You only change typeid if the RVal is empty, but in L1239, you may change the meta as long as the member length is not equal, so there may still be circumstances where the assigned typeid is not matched with the actual meta

Reply:
done